packages/asset-service/src/service/AssetService.ts

```diff
@@ -12,15 +18,45 @@ export const flattenAssetData = (assetData: BaseAsset[]): Asset[] => {
     flatAssetData.push(newAsset)
     if (baseAsset.tokens) {
       for (const tokenAsset of baseAsset.tokens) {
-        flatAssetData.push({
+        const flatAssetDatum = {
           ...tokenAsset,
           chain: baseAsset.chain,
           network: baseAsset.network,
           slip44: baseAsset.slip44,
           explorer: baseAsset.explorer,
           explorerAddressLink: baseAsset.explorerAddressLink,
           explorerTxLink: baseAsset.explorerTxLink
-        })
+        }
+        if (typeof flatAssetDatum.icon === 'undefined') {
+          const { tokenId } = caip19.fromCAIP19(flatAssetDatum.caip19)
+          const symbolPixels = renderPixels(flatAssetDatum.symbol[0], fonts.sevenPlus) // 7 x 5
+          const symbolScale = 12
+          const imageSize = 128
+          const offsetX = Math.round((imageSize - 5 * symbolScale) / 2)
+          const offsetY = Math.round((imageSize - 7 * symbolScale) / 2)
+          const options = {
+            background: [45, 55, 72, 255],
+            size: imageSize
+          }
+          const identiconImage = new Identicon(tokenId?.substring(2), options).render()
+          // render pixel symbol
+          for (let x = 0; x < imageSize; x++) {
+            const offsettedX = Math.floor((x - offsetX) / symbolScale)
+            if (offsettedX < 0 || offsettedX >= 5) continue
+            for (let y = 0; y < imageSize; y++) {
+              const offsettedY = Math.floor((y - offsetY) / symbolScale)
+              if (offsettedY < 0 || offsettedY >= 7) continue
+              if (symbolPixels[offsettedY][offsettedX]) {
+                identiconImage.buffer[identiconImage.index(x, y)] = identiconImage.color.apply(
```

Comment:
Isn't there any way to make this work without fiddling with the internals of identicon's `pngLib`?
This is not documented and will make it harder to maintain.